File: internal/user/schedule.go
package user

// import (
// 	"time"
// )

// DailySchedule represents a collection of timeboxed activities that a user
// has determined to accomplish.
type DailySchedule struct {
	Activities []ScheduledActivity `json:"activities" bson:"activities"`
}

// ScheduledActivity represents a singular timeboxed activity that a user wants
// to complete.
type ScheduledActivity struct {
	Start       string `json:"start" bson:"start"`
	End         string `json:"end" bson:"end"`
	Title       string `json:"title" bson:"title"`
	Description string `json:"description" bson:"description"`
}

File: api-python/src/schedule.py
from typing import List
from pydantic import BaseModel, Field

class ScheduledActivity(BaseModel):
    """A singular timeboxed activity that a user wants to complete"""
    title: str 
    description: str
    startTime: str
    endTime: str

class DailySchedule(BaseModel):
    """A daily collection of timeboxed activities that a user wants to complete"""
    routine: List[ScheduledActivity]

